module: refactor gateway script to unify checkers

leemgs opened the pull request:
Fixed issue #299.

This commit is to refator a gateway script to unify the existing
two checkers such as format and audit.

**Changes proposed in this PR:**

* Version 2:
1. Refactor the gateway script after merging two check scripts
2. Unified the report folder consistently
3. Added annotations for maintenance

* Version 1:
1. Unified start-up code of the existing two checker
2. Removed duplicated statements
3. Merged two folders to one folder, pr-checker

Signed-off-by: Geunsik Lim <leemgs@gmail.com>

leemgs commented:
The aging test is completed.

ohsewon commented on ci/taos/checker-pr-gateway.sh:
```diff
@@ -31,27 +31,99 @@
+echo -e "[DEBUG] Running"                                            | tee -a $log_file
+echo -e "[DEBUG] ......"                                             | tee -a $log_file
+echo -e "[DEBUG] Completed"                                          | tee -a $log_file
+# --------------------------- Run module: audit ---------------------------------------------------------------
```
Does run audit module after finishing format module?

leemgs replied:
Updated.

ohsewon commented on ci/taos/checker-pr-gateway.sh:
```diff
@@ -31,27 +31,99 @@
+cd $dir_ci
+dir_worker="repo-workers/pr-checker"
+export dir_worker=$dir_worker
+export dir_commit=${dir_worker}/${input_date}-${input_pr}-${input_commit}
```
100~105 lines are duplicated with 69~74

ohsewon commented on ci/taos/checker-pr-gateway.sh:
```diff
@@ -31,27 +31,99 @@
+    echo "[DEBUG] WARN: mkdir command is failed because $dir_commit directory already exists."
+    echo "[DEBUG] WARN: So removing the existing directory..."
+    rm -rf ./${dir_commit}
+fi
```
107~112 need to be removed.

ohsewon commented on ci/taos/checker-pr-gateway.sh:
```diff
@@ -31,27 +31,99 @@
+
+
+# Save a log file for debugging
+mkdir -p $dir_commit
```
this line is unnecessary

ohsewon commented on ci/taos/checker-pr-gateway.sh:
```diff
@@ -31,27 +31,99 @@
+echo -e "[DEBUG] current path: $(pwd)."                              | tee -a $log_file
+echo -e "[DEBUG] ./checker-pr-audit-async.sh $1 $2 $3 $4 $5 $6 "     | tee -a $log_file
+echo -e "[DEBUG] Starting asynchronouly...               "           | tee -a $log_file
+cd ./taos/
```
IMO, this line will make error

leemgs commented:
**Changes proposed in this PR:**

* Version 2:
1. Refactor the gateway script after merging two check scripts
2. Unified the report folder consistently

* Version 1:
1. Unified start-up code of the existing two checker
2. Removed duplicated statements
3. Merged two folders to one folder, pr-checker

leemgs commented:
**Changes proposed in this PR:**

* Version 2:
1. Refactor the gateway script after merging two check scripts
2. Unified the report folder consistently
3. Added annotations for maintenance

* Version 1:
1. Unified start-up code of the existing two checker
2. Removed duplicated statements
3. Merged two folders to one folder, pr-checker
